### PR TITLE
Update accessible roles for v5

### PR DESCRIPTION
**Pull request by clarkgunn**
### Description

Adjusts roles and attributes to better align with the WCAG Aria Authoring Practices Guide https://www.w3.org/WAI/ARIA/apg/patterns/carousel

* New `tabbed` prop determines whether roles should follow the Basic Carousel or Tabbed Carousel requirements.
* New `landmark` prop determines whether role should be a `region` landmark or have role `group`. Role and label have been moved from the slider frame to the parent carousel element so controls are contained within the label and role. Landmark regions should be intentional, so the default value is set to false.
* New `carouselId` prop gives the carousel frame an `id` attribute which can be uniquely targeted by controls where there might be multiple carousels on a page. This could be replaced internally with `useId` with React 18.
* Default `frameAriaLabel` has been changed to a slightly more useful 'Slider', the role description of 'carousel'  will already be announced 
* Default pagination dots have been given the role `tab`, and `aria-controls` targeting their slide.
* Pagination dots parent has been given the role `tablist`. As `tab`s need to be the immediate child of `tablist`, and since `tablist` already has list semantics, `ul` and `li` have been removed. This should resolve https://github.com/FormidableLabs/nuka-carousel/issues/1001 and the issue raised in https://github.com/FormidableLabs/nuka-carousel/pull/803 regarding `aria-selected`.
* When `tabbed` is true, slides will have the role `tabpanel` indicating they are the target of the `tab`, otherwise they will have the role `group` and a description of `slide`.
* When `tabbed` is false, default pagination dots will not be rendered.
* Default next and previous buttons now target the slider frame with `aria-controls`.

Adds dev dependencies: `jest-axe`, `@types/jest-axe`

Fixes https://github.com/FormidableLabs/nuka-carousel/issues/1001

#### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

These changes will impact the roles and attributes when default props are used. I believe this will improve accessibility but may be considered a breaking change.

### How Has This Been Tested?

New jest and cypress tests were added. Existing cypress tests needed wait logic to pass, akin to this change https://github.com/FormidableLabs/nuka-carousel/pull/887/commits/1f7f0200e5b5fe45f0f55379d4c909836fc2d5cb

Cypress and Jest tests and Lighthouse accessibility report should all pass. Attributes and roles can be verified in devtools.

A tabbed carousel with chrome devtools displaying accessible roles and attributes.

An untabbed region landmark carousel with chrome devtools displaying accessible roles and attributes.

### Checklist

- [x] My code follows the style guidelines of this project (I have run `pnpm run lint`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes (I have run `pnpm run test:ci-with-server`/`pnpm run test`)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have recorded any user-facing fixes or changes with `pnpm changeset`.
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/nuka/src/carousel.test.tsx [hunk 1]
```diff
@@ -140,39 +158,45 @@ describe('Carousel', () => {
...
       keyCodeConfig,
       slideCount,
       beforeSlide,
+      frameAriaLabel: 'keyboard',
+      landmark: true,
+      carouselId,
     });
 
-    const carouselFrame = screen.getByRole('region');
+    const sliderFrame = screen.getByTestId(`${carouselId}-slider-frame`);
```
Comment (clarkgunn) [inline packages/nuka/src/carousel.test.tsx:171]: `frameAriaLabel` now references the whole carousel, this ensures the same slider frame element is being selected.

File: packages/nuka/src/types.ts [hunk 2]
```diff
@@ -222,6 +225,11 @@ export interface InternalCarouselProps {
    */
   beforeSlide: (currentSlideIndex: number, endSlideIndex: number) => void;
 
+  /**
+   * Unique id attribute for the carousel which may be referenced by aria attributes.
+   */
+  carouselId: string;
```
Comment (clarkgunn) [inline packages/nuka/src/types.ts:231]: I went with `carouseld` to disambiguate between `id` attributes and props on other elements.

File: packages/nuka/src/types.ts [hunk 3]
```diff
@@ -428,6 +441,11 @@ export interface InternalCarouselProps {
    */
   swiping: boolean;
 
+  /**
+   * Whether tab pagination is used to set appropriate roles for slides.
+   */
+  tabbed: boolean;
```
Comment (clarkgunn) [inline packages/nuka/src/types.ts:447]: `tabbed` doesn't exactly align with the "pagination" and "dots" verbiage currently used, but it aligns with WCAG terminology.

File: packages/nuka/cypress/integration/carousel.spec.ts [hunk 4]
```diff
@@ -1,4 +1,5 @@
 /* eslint-disable cypress/no-unnecessary-waiting */
+const waitThreshold = 300;
```
Comment (clarkgunn) [inline packages/nuka/cypress/integration/carousel.spec.ts:2]: http://localhost:3000/?slides=4&params=%7B%22landmark%22:true,%22tabbed%22:false%7D

File: packages/nuka/src/default-controls.tsx [hunk 5]
```diff
@@ -268,8 +274,15 @@ export const PagingDots = ({
...
   const currentSlideBounded = getBoundedIndex(currentSlide, slideCount);
 
+  if (!tabbed) return null;
+
   return (
-    <ul className={pagingDotsContainerClassName} style={listStyles}>
+    <div
+      className={pagingDotsContainerClassName}
+      style={listStyles}
+      aria-label="Choose slide to display."
```
Comment (carloskelly13) [inline packages/nuka/src/default-controls.tsx:283]: Should we let this be customizable with this string as default? (maybe this is more for an eventual port of this for v6 and not this specific use case)

**Review by carloskelly13 — APPROVED** [review]
Thank you for these improvements and getting this more inline. This will be also great to forward-port to v6 as well.

**Pull request merged**